core/src/main/java/com/gentics/mesh/core/endpoint/node/NodeCrudHandler.java

```diff
@@ -163,9 +161,9 @@ public void handleMove(InternalActionContext ac, String uuid, String toUuid) {
 	public void handleNavigation(InternalActionContext ac, String uuid) {
 		validateParameter(uuid, "uuid");
 
-		utils.rxSyncTx(ac, tx -> {
+		utils.syncTx(ac, tx -> {
 			Node node = getRootVertex(ac).loadObjectByUuid(ac, uuid, READ_PERM);
-			return node.transformToNavigation(ac);
+			return node.transformToNavigation(ac).blockingGet();
 		}, model -> ac.send(model, OK));
```

Comment:
subscribe is non-blocking and will thus conflict with the way the locking mechanism works.

```java
	@Test
	public void testSubLock() throws InterruptedException {
		Completable s = Completable.create(e -> {
			Thread.sleep(1000);
			System.out.println("Done");
			e.onComplete();
		});

		System.out.println("Lock Start");
		s.subscribeOn(Schedulers.newThread()).subscribe(() -> {
			System.out.println("Done");
		}, err -> {
			System.err.println(err.getMessage());
		});
		System.out.println("Lock End");
		Thread.sleep(2000);
	}
```

As you can see it the execution of the completable happens outside of the timeframe in which the lock is held. Thus the code needs to be blocking. I know that it would be possible to use onXYZ methods to handle the lock and release it but I don't trust this enough. I think a regular try-with-resource is the safer option.

Reply:
As mentioned in our call.. I did not realize that this code does not need a lock. Thus asyncTx works fine here. I changed the code.